spring-integration-core/src/main/java/org/springframework/integration/handler/LoggingHandler.java

```diff
@@ -164,58 +169,66 @@ protected void onInit() throws Exception {
 	@Override
 	protected void handleMessageInternal(Message<?> message) throws Exception {
 		switch (this.level) {
-		case FATAL:
-			if (this.messageLogger.isFatalEnabled()) {
-				this.messageLogger.fatal(createLogMessage(message));
-			}
-			break;
-		case ERROR:
-			if (this.messageLogger.isErrorEnabled()) {
-				this.messageLogger.error(createLogMessage(message));
-			}
-			break;
-		case WARN:
-			if (this.messageLogger.isWarnEnabled()) {
-				this.messageLogger.warn(createLogMessage(message));
-			}
-			break;
-		case INFO:
-			if (this.messageLogger.isInfoEnabled()) {
-				this.messageLogger.info(createLogMessage(message));
-			}
-			break;
-		case DEBUG:
-			if (this.messageLogger.isDebugEnabled()) {
-				this.messageLogger.debug(createLogMessage(message));
-			}
-			break;
-		case TRACE:
-			if (this.messageLogger.isTraceEnabled()) {
-				this.messageLogger.trace(createLogMessage(message));
-			}
-			break;
-		default:
-			throw new IllegalStateException("Level '" + this.level + "' is not supported");
+			case FATAL:
+				if (this.messageLogger.isFatalEnabled()) {
+					this.messageLogger.fatal(createLogMessage(message));
+				}
+				break;
+			case ERROR:
+				if (this.messageLogger.isErrorEnabled()) {
+					this.messageLogger.error(createLogMessage(message));
+				}
+				break;
+			case WARN:
+				if (this.messageLogger.isWarnEnabled()) {
+					this.messageLogger.warn(createLogMessage(message));
+				}
+				break;
+			case INFO:
+				if (this.messageLogger.isInfoEnabled()) {
+					this.messageLogger.info(createLogMessage(message));
+				}
+				break;
+			case DEBUG:
+				if (this.messageLogger.isDebugEnabled()) {
+					this.messageLogger.debug(createLogMessage(message));
+				}
+				break;
+			case TRACE:
+				if (this.messageLogger.isTraceEnabled()) {
+					this.messageLogger.trace(createLogMessage(message));
+				}
+				break;
+			default:
+				throw new IllegalStateException("Level '" + this.level + "' is not supported");
 		}
 	}
 
-
 	private Object createLogMessage(Message<?> message) {
+		Assert.notNull(this.expression, "'expression' must not be null");
```

Comment:
This is fine during initialization, but sounds strange at runtime.
The `notNull()` throws `IllegalArgumentException` which doesn't reflect reality for the current `message`. 
We should use here `Assert.state()`, because it is really `IllegalStateException` during handling `message`.

OTOH I have changed my mind and decided to have this code after one more review:
```
private volatile Expression expression = EXPRESSION_PARSER.parseExpression("payload");
...
private volatile EvaluationContext evaluationContext = ExpressionUtils.createStandardEvaluationContext();
```
This isn't Spring Context aware, but lets do not have these asserts at all. Plus such a change reinstates the previous behavior. Therefore the next `4.3.5` will let you to upgrade without issue here :smile: